nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/queue/TestGetAzureQueueStorage_v12.java

```diff
@@ -0,0 +1,143 @@
+        runner.setProperty(GetAzureQueueStorage_v12.STORAGE_CREDENTIALS_SERVICE, CREDENTIALS_SERVICE_IDENTIFIER);
+    }
+
+    @Test
+    public void testValidVisibilityTimeout() {
+        runner.setProperty(GetAzureQueueStorage_v12.QUEUE, "dummyqueue");
+        runner.setProperty(GetAzureQueueStorage_v12.BATCH_SIZE, "10");
+        runner.setProperty(GetAzureQueueStorage_v12.VISIBILITY_TIMEOUT, "10 secs");
+
+        ProcessContext processContext = runner.getProcessContext();
+        Collection<ValidationResult> results = new HashSet<>();
+        if (processContext instanceof MockProcessContext) {
+            results = ((MockProcessContext) processContext).validate();
+        }
+
+        assertEquals(0, results.size());
+    }
+
+    @Test
+    public void testInvalidVisibilityTimeoutZeroSecs() {
+        runner.setProperty(GetAzureQueueStorage_v12.QUEUE, "dummyqueue");
+        runner.setProperty(GetAzureQueueStorage_v12.BATCH_SIZE, "10");
+        runner.setProperty(GetAzureQueueStorage_v12.VISIBILITY_TIMEOUT, "0 secs");
+
+        ProcessContext processContext = runner.getProcessContext();
+        Collection<ValidationResult> results = new HashSet<>();
+        if (processContext instanceof MockProcessContext) {
+            results = ((MockProcessContext) processContext).validate();
+        }
+
+        assertEquals(1, results.size());
+        Iterator<ValidationResult> iterator = results.iterator();
+        assertTrue(iterator.next().getExplanation().contains("0 secs"));
+    }
+
+    @Test
+    public void testInvalidVisibilityTimeoutMoreThanSevenDays() {
+        runner.setProperty(GetAzureQueueStorage_v12.QUEUE, "dummyqueue");
+        runner.setProperty(GetAzureQueueStorage_v12.BATCH_SIZE, "10");
+        runner.setProperty(GetAzureQueueStorage_v12.VISIBILITY_TIMEOUT, "8 days");
+
+        ProcessContext processContext = runner.getProcessContext();
+        Collection<ValidationResult> results = new HashSet<>();
+        if (processContext instanceof MockProcessContext) {
+            results = ((MockProcessContext) processContext).validate();
+        }
+
+        assertEquals(1, results.size());
+        Iterator<ValidationResult> iterator = results.iterator();
+        assertTrue(iterator.next().getExplanation().contains("7 days"));
+    }
+
+    @Test
+    public void testValidRequestTimeout() {
+        runner.setProperty(GetAzureQueueStorage_v12.QUEUE, "dummyqueue");
+        runner.setProperty(GetAzureQueueStorage_v12.BATCH_SIZE, "10");
+        runner.setProperty(GetAzureQueueStorage_v12.VISIBILITY_TIMEOUT, "10 secs");
+        runner.setProperty(GetAzureQueueStorage_v12.REQUEST_TIMEOUT, "15 secs");
+
+        ProcessContext processContext = runner.getProcessContext();
+        Collection<ValidationResult> results = new HashSet<>();
+        if (processContext instanceof MockProcessContext) {
+            results = ((MockProcessContext) processContext).validate();
+        }
```

Comment:
The `TestRunner.assertValid()` method can be used instead of this approach.

Reply:
Making the changes